@vates/types/src/common.mts

```diff
@@ -0,0 +1,638 @@
+declare const __brand: unique symbol
+
+export type Branded<TBrand extends string, TType = string> = TType & { [__brand]: TBrand }
+
+export const TASK_ALLOWED_OPERATIONS = {
+  CANCEL: 'cancel',
+  DESTROY: 'destroy',
+} as const
+
+export type TASK_ALLOWED_OPERATIONS = (typeof TASK_ALLOWED_OPERATIONS)[keyof typeof TASK_ALLOWED_OPERATIONS]
+
+export const TASK_STATUS_TYPE = {
+  CANCELLED: 'cancelled',
+  CANCELLING: 'cancelling',
+  FAILURE: 'failure',
+  PENDING: 'pending',
+  SUCCESS: 'success',
+} as const
+
+export type TASK_STATUS_TYPE = (typeof TASK_STATUS_TYPE)[keyof typeof TASK_STATUS_TYPE]
```

Comment:
do I understand this correctly ?

this instantiate a type that mirror the map before and will only accept the key of the map ?